src/composables/percentageTool.ts

```diff
@@ -1,8 +1,9 @@
 import { ElementEvent } from 'echarts';
 import { ROUND_CLOSER, roundTimeframe } from '@/shared/timemath';
 import humanizeDuration from 'humanize-duration';
+import { CandleChartProps } from '@/types';
 
-export function usePercentageTool(chartRef, props) {
+export function usePercentageTool(chartRef, props: CandleChartProps) {
```

Comment:
well an entirely different topic might be if we really need or want to pass in the whole "props" object, or just the 2 settings we use (theme, and timeframe, iirc).

I didn't do at the time as i wasn't sure what's going to be needed - but i think that might be the better option - unless you think that has some downsides i didn't consider.

Reply:
@Tako88 what i mean is essentially something around what i did in #1634

Reply:
That was a consideration for me at some point, but we need reactivity for the theme switch, or will have to handle it manually in some place. I did try using toRef to pass it in, but there was some weird behavior, I can't remember what exactly rn, decided to just go for the whole props object because I knew that just worked.

Reply:
There's 2 things i don't like with this approach - it's dragging the actual type away from the component - and hard-links the composable to this particular chart, neither of which i do like.

i've tried with to-refs now - and it seems to work just fine - at least as far as i can see.

Reply:
Hm okay, tried it now again as well and it works fine, I think I didn't do ```toRef(() => props.theme)```maybe last time and just passed props.theme directly. 

So in that case I agree this is the better approach. It is a couple lines worth of changes, do you want a PR or just do them yourself?

Reply:
it's available as PR in #1634 ... ;) Wanted you to take a look at first :)

Reply:
> it's available as PR in #1634 ... ;) Wanted you to take a look at first :)

Ooh sorry, I don't know how I missed that.